code/modules/vehicles/vehicle_key.dm

```diff
@@ -4,6 +4,42 @@
 	icon = 'icons/obj/vehicles.dmi'
 	icon_state = "key"
 	w_class = WEIGHT_CLASS_TINY
+	/// You can only do a big jingle that affects people this often, otherwise it just plays the jingle sound more quietly
+	var/big_jingle_delay = 5 SECONDS
+	/// The cooldown for big jingles
+	COOLDOWN_DECLARE(big_jingle_cd)
+
+/obj/item/key/attack_self(mob/user, modifiers)
+	if(!COOLDOWN_FINISHED(src, big_jingle_cd))
+		to_chat(user, "<span class='smallnotice'>You jingle the keys in your hand softly...</span>")
+		playsound(src, pick('sound/items/key_jingle01.ogg', 'sound/items/key_jingle02.ogg'), 30)
+		return
+
+	. = ..()
+
+	COOLDOWN_START(src, big_jingle_cd, big_jingle_delay)
+	user.visible_message("<span class='notice'>[user] jingles [user.p_their()] keys around in an enticing fashion.</span>", "<span class='notice'>You jingle the keys around in an enticing fashion.</span>")
+	playsound(src, pick('sound/items/key_jingle01.ogg', 'sound/items/key_jingle02.ogg'), 60)
+
+	for(var/mob/living/iter_living in view(5, get_turf(user)))
+		if(iter_living.stat > CONSCIOUS)
+			continue
+
+		if(prob(50) && HAS_TRAIT(iter_living, TRAIT_DUMB)) // 50% chance to freeze those with simple minds
+			iter_living.visible_message("<span class='danger'>[iter_living] freezes in [iter_living.p_their()] tracks, transfixed by the jingling keys...</span>",\
+				"<span class='userdanger'>Is that-... are those... jingle jingle jingle... beautiful...</span>", vision_distance = COMBAT_MESSAGE_RANGE)
+			iter_living.Stun(rand(1 SECONDS, 3 SECONDS))
+		else if(prob(20) && (isfelinid(iter_living) || ismoth(iter_living)))
```

Comment:
I'd personally prefer for this to be based on ear protection instead of a die roll, so that you can more consistently pull off the joke when it'd be funny to do it demands it (jingling keyes for like 25 seconds straight to try to trigger a pounce seems like it'd ruin the comedic timing), but not be able to use it to cheese out a kill against a sec officer or something.